### PR TITLE
boards: Remove the unnecessary customize Make.defs

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since the general Make.defs in scripts folder is enough

## Impact
No

## Testing
Pass CI

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: boards/arm/stm32/stm32f4discovery/scripts/Make.defs [hunk 1]
```diff
@@ -35,7 +35,11 @@ ifneq ($(CONFIG_DEBUG_NOOPT),y)
 endif
 
 ARCHCFLAGS = -fno-builtin
-ARCHCXXFLAGS = -fno-builtin -fno-exceptions -fcheck-new
+ifeq ($(CONFIG_CXX_EXCEPTION),y)
+  ARCHCXXFLAGS = -fno-builtin
+else
+  ARCHCXXFLAGS = -fno-builtin -fno-exceptions -fcheck-new
```
Comment (pkarashchenko) [inline boards/arm/stm32/stm32f4discovery/scripts/Make.defs:41]: Why we do not have `-fno-rtti` here?

Comment (xiaoxiang781216) [reply boards/arm/stm32/stm32f4discovery/scripts/Make.defs:41]: Add at line 52:
https://github.com/apache/incubator-nuttx/pull/5555/files#diff-0a874401241f632c07d97d166345317a75e439db54eb33b3082394ba2a6749bbL48

File: boards/arm/imxrt/imxrt1060-evk/configs/libcxxtest/Make.defs [hunk 2]
```diff
@@ -1,80 +0,0 @@
...
-############################################################################
-
-include $(TOPDIR)/.config
-include $(TOPDIR)/tools/Config.mk
-include $(TOPDIR)/arch/arm/src/armv7-m/Toolchain.defs
-
-ifeq ($(CONFIG_ARMV7M_DTCM),y)
-  LDSCRIPT = flash-dtcm.ld
-else
-  LDSCRIPT = flash-ocram.ld
```
Comment (davids5) [inline boards/arm/imxrt/imxrt1060-evk/configs/libcxxtest/Make.defs:28]: @xiaoxiang781216 Where is this functionality now?

**Pull request merged**